lib/rack/test/methods.rb

```diff
@@ -55,7 +55,7 @@ def current_session
       # Create a new session (or reuse an existing session with the given name),
       # and make it the current session for the given block.
       def with_session(name)
-        session = @_rack_test_current_session
+        session = @_rack_test_current_session if defined?(@_rack_test_current_session)
```

Comment:
Can we initialize this to nil in `initialize` rather than using `defined?`?

Reply:
This module is included an arbitrary classes, and we don't know how their `initialize` method is defined. Unless the class's `initialize` method calls super, `initialize` in this module will not be called.  Even if we assume `initialize` in the including class calls `super`, we would have to override with `def initialize(*args, &block)`, and use `ruby2_keywords`, and that would still change the arity of `initialize`.  I don't think that approach is a good idea.

Reply:
Another option is to use attr_accessor which handles not being initialized.

Reply:
private attr_accessor would also work, let me know if you prefer that I switch to that.

Reply:
Yes, I think it's better.

Reply:
OK, switched to private attr_accessor